avocado/core/jobdata.py

```diff
@@ -146,6 +146,26 @@ def retrieve_variants(resultsdir):
     """
     Retrieves the job Mux object from the results directory.
     """
+    def _apply_workarounds(variants):
+        """
+        Older version of TreeNode used 2 dicts to store environment, lets
+        replace the dict with TreeEnvironment to make it work properly
+        from AvocadoParams point of view.
+        """
+        try:
+            for plugin in variants._variant_plugins.extensions:
+                if plugin.name == "yaml_to_mux":
+                    for variant in plugin.obj.variants.pools:
+                        for mux_tree in variant:
+                            for node in mux_tree.pools:
+                                if hasattr(node, "filters"):
+                                    return  # it's new (or renewed) style
+                                root = node.root
+                                for _node in root.iter_children_preorder():
+                                    _node.filters = [[], []]
+                                    _node._environment = None
+        except:
+            pass
```

Comment:
Ok, hack-ish, but functional. I think we should clearly mark/comment those compatibility hacks so we can re-evaluate and possible remove them when we consider appropriate.